src/apps/ocioview/ocioview/inspect/cube_inspector.py

```diff
@@ -0,0 +1,789 @@
+        """
+        Set the ratio of pixels sampled from the current image.
+
+        :param image_detail: Ratio of image samples to pixels
+        """
+        if image_detail != self._image_detail:
+            self._image_detail = image_detail
+            self._update_cube()
+
+    def set_show_source_cube(self, show: bool) -> None:
+        """
+        Set whether to draw the source cube edges in the scene.
+
+        :param show: Whether to show source cube
+        """
+        if show != self._show_src_cube:
+            self._show_src_cube = show
+            self._update_cube()
+
+    def set_clamp_to_source_cube(self, clamp: bool) -> None:
+        """
+        Set whether to clamp samples that fall outside the source cube.
+
+        :param clamp: Whether to clamp samples tp source cube
+        """
+        if clamp != self._clamp_to_src:
+            self._clamp_to_src = clamp
+            self._update_cube()
+
+    def fit(self) -> None:
+        """Fit cube to viewport."""
+        camera = self._window.camera()
+        camera.viewEntity(self.lines_entity)
+
+    def _update_point_size(self):
+        """Update point size relative to camera distance."""
+        camera = self._window.camera()
+        camera_dist = camera.position().distanceToPoint(camera.viewCenter())
+
+        # Dynamic point size. Linearly map [100, 0] camera distance range to
+        # [1, 10] point size range.
+        amount = min(1.0, max(0.0, (100 - camera_dist) / 100))
+        point_size = 1 + (10 - 1) * amount
+        self.point_size.setValue(point_size)
+
+    def _update_picker_tolerance(self):
+        """
+        Update point picker tolerance relative to cube size, providing
+        each point a hit area adjacent to its neighboring points.
+        """
+        #   o-----o-----o    cube_size points
+        #   o--o--o--o--o    cube_size+1 sections
+        # --o--|--o--|--o--  tolerance == 2x section length, centered on vertex
+        self.picking_settings.setWorldSpaceTolerance((10.0 / (self._cube_size + 1)) * 2)
+
+    def _update_cube(self) -> None:
+        """
+        Update cube point cloud and lines from parameters and transform
+        data, if available.
+        """
+        # Points
+        if self._input_src == InputSource.IMAGE and self._image_buf is not None:
+            image_spec = self._image_buf.spec()
+            identity_grid = self._image_buf.get_pixels(
+                oiio.FLOAT,
+                roi=oiio.ROI(
+                    # fmt: off
+                    image_spec.x,
+                    image_spec.x + image_spec.width,
+                    image_spec.y,
+                    image_spec.y + image_spec.height,
+                    0, 1,
+                    0, 3,
+                    # fmt: on
+                ),
+            ).reshape((image_spec.width * image_spec.height, 3))[:: self._image_detail]
+            sample_count = identity_grid.size // 3
```

Comment:
I haven't looked at all the code yet but as I'm working on the chromaticity diagram and I need a `ndarray`, I'm wondering if rather than passing an OIIO `ImageBuf` around it would not be better to be in `Numpy` land.

Reply:
Agree with that, in #1853 I'm making OIIO an optional dependency (same for Imath, basically everything that doesn't have PyPI package ideally).

Reply:
Good point. I can change it to a `ndarray`.